src/customParser.ts

```diff
@@ -0,0 +1,330 @@
+          this.currentValue = 0;
+          this.nextFibIndex = 0;
+          this.readBitId++;
+          return returnValue;
+        } else { // Otherwise update read value
+          this.currentValue += lookup[this.nextFibIndex];
+        }
+      }
+      this.nextFibIndex++;
+      this.readBitId++;
+    }
+    throw Error('THIS IS REALLY BAD');
+  }
+}
+
+class BufferReader {
+  public dataView: DataView;
+  public location: number;
+
+  private arrayBuffer: ArrayBuffer;
+  private flip: boolean;
+
+  constructor(arrayBuffer: ArrayBuffer, flip: boolean) {
+    this.arrayBuffer = arrayBuffer;
+    this.dataView = new DataView(arrayBuffer);
+    this.location = 0;
+    // FLIP ONLY IF READING A NODE FILE
+    this.flip = Boolean(flip);
+  }
+
+  readUint32() {
+    this.location += 4;
+    return this.dataView.getUint32(this.location - 4, this.flip);
+  }
+
+  readUint32Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getUint32(this.location + i * 4, this.flip));
+  }
+  this.location += numberOfValues * 4;
+  return newArray;
+  }
+
+  readUint16Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getUint16(this.location + i * 2, this.flip));
+  }
+  this.location += numberOfValues * 2;
+  return newArray;
+  }
+
+  readUint8Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getUint8(this.location + i));
+  }
+  this.location += numberOfValues;
+  return newArray;
+  }
+
+  readFloat32Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getFloat32(this.location + i * 4, this.flip));
+  }
+  this.location += numberOfValues * 4;
+  return newArray;
+  }
+
+  skip(numberOfBytes: number) {
+    this.location += numberOfBytes;
+  }
+}
+
+function parseCustomFileFormat(asArrayBuffer: ArrayBuffer, flip: boolean) {
+  const fileReader = new BufferReader(asArrayBuffer, true);
+
+  const infoHeader = fileReader.readUint32Array(4);
+  const magicBytes = infoHeader[0];
+  const formatVersion = infoHeader[1];
+  const optimizerVersion = infoHeader[2];
+  const arrayCount = infoHeader[3];
+
+  const arraysList = ['colors', 'x_values', 'y_values', 'z_values', 'normals', 'deltas', 'heights', 'radiuses',
+  'angles', 'matrixes', 'x_translations', 'y_translations', 'z_translations'];
+  const arraysDictionary: any = {};
+  for (let arrayId = 0; arrayId < arrayCount;  arrayId++) {
+    const clusterCount = fileReader.readUint32();
+    const bytesForOneValue = fileReader.readUint32();
+    const type = arraysList[arrayId];
+
+    if ((type === 'colors') && (bytesForOneValue === 4)) {
+      const colorValues = fileReader.readUint8Array(clusterCount * 4);
+      arraysDictionary[type] = [];
```

Comment:
Do we know the size of this array so we could use fixed size Int8Array or something?